publications-by-owner/src/test/java/no/unit/nva/publication/owner/PublicationsByOwnerHandlerTest.java

```diff
@@ -0,0 +1,170 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.util.Collections.singletonMap;
+import static no.unit.nva.PublicationHandler.ACCESS_CONTROL_ALLOW_ORIGIN;
+import static no.unit.nva.PublicationHandler.ALLOWED_ORIGIN_ENV;
+import static no.unit.nva.model.PublicationStatus.DRAFT;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.HttpStatus.SC_BAD_GATEWAY;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PublicationsByOwnerHandlerTest {
+
+    public static final String OWNER = "junit";
+    public static final String VALID_ORG_NUMBER = "NO919477822";
+    private ObjectMapper objectMapper = PublicationHandler.createObjectMapper();
+
+    private Environment environment;
+    private PublicationService publicationService;
+    private Context context;
+
+    private OutputStream output;
+    private PublicationsByOwnerHandler publicationsByOwnerHandler;
+
+    /**
+     * Set up environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        environment = mock(Environment.class);
+        when(environment.get(ALLOWED_ORIGIN_ENV)).thenReturn(Optional.of("*"));
+
+        publicationService = mock(PublicationService.class);
+        context = mock(Context.class);
+
+        output = new ByteArrayOutputStream();
+        publicationsByOwnerHandler =
+                new PublicationsByOwnerHandler(objectMapper, publicationService, environment);
+
+    }
+
+    @Test
+    @DisplayName("default Constructor Throws Exception When Envs Are Not Set")
+    public void defaultConstructorThrowsExceptionWhenEnvsAreNotSet() {
+        assertThrows(Exception.class, () -> new PublicationsByOwnerHandler());
+    }
+
+    @Test
+    @DisplayName("handler Returns Ok Response On Valid Input")
+    public void handlerReturnsOkResponseOnValidInput() throws IOException, InterruptedException {
+        when(publicationService.getPublicationsByOwner(anyString(), any(URI.class), any()))
+                .thenReturn(publicationSummaries());
+
+        publicationsByOwnerHandler.handleRequest(
+                inputStream(), output, context);
+
+        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        assertEquals(SC_OK, gatewayResponse.getStatusCode());
+        Assert.assertTrue(gatewayResponse.getHeaders().keySet().contains(CONTENT_TYPE));
+        Assert.assertTrue(gatewayResponse.getHeaders().keySet().contains(ACCESS_CONTROL_ALLOW_ORIGIN));
+    }
+
+    @Test
```

Comment:
What happens when the doi is valid but non-existent? i.e., if we receive a 404 from the server?

Reply:
What do you mean?